provider/core/src/response.rs

```diff
@@ -362,9 +384,16 @@ where
             PhantomData<&'a ()>,
         ) -> <M2::Yokeable as Yokeable<'a>>::Output,
     {
-        DataPayload {
-            yoke: self.yoke.map_project_cloned(f),
-        }
+        DataPayload(DataPayloadInner::Yoke(match &self.0 {
+            DataPayloadInner::Yoke(yoke) => yoke.map_project_cloned(f),
+            DataPayloadInner::Ref(r) => {
+                let output: <M2::Yokeable as Yokeable<'static>>::Output =
+                    f(Yokeable::transform(*r), PhantomData);
+                // Safety: <M2::Yokeable as Yokeable<'static>>::Output is the same type as M2::Yokeable
```

Comment:
Suggestion: just downgrade the thing to a Yoke via ZeroFrom instead of doing these unsafe gymnastics

Reply:
Why? `f` operates on a reference so we can use that to avoid doing the expensive ZeroFrom which we're trying to avoid with this PR. It's not gymnastics if it's correct.

Reply:
Ok, well there's still necessarily a `Clone` or `ZeroFrom` happening inside of the `f` function, but the advantage is that in `map_project_cloned` we only need to do that on a subset of fields instead of the whole thing.

I think the safety comment is not quite right; the invariant is "The returned value must be destroyed before the data from was borrowing from is." The fact that you're setting `'a` to `'static` on line 390 is important to the safety of the borrowed data.

Reply:
I think "The returned value must be destroyed before the data from was borrowing from is." is irrelevant. The point here is that the types are the same but the compiler doesn't know, I could also use a transmute.

Reply:
Well I can't use a transmute because the compiler doesn't know they're the same size

Reply:
I would ideally like to avoid having any unsafety here.

I _think_ all we need here is `fn upgrade_output<Y: Yokeable<'static>>(output: Y::Output) -> Y`, which internally calls `Yokeable::make()`. Does that make sense? `Yokeable::make()` is only scarily unsafe when the lifetimes involved aren't `'static`.